### PR TITLE
Feat: add getFiles selector

### PR DESCRIPTION
**Pull request by lolottetheclash**
The purpose of this PR is to add a selector in order to recover all the loaded files

### CODE REVIEW

**Review by JoeyDjr — APPROVED** [review]

**Review by ErikssonJoakim — APPROVED** [review]
Cool stuff 👍

File: src/domain/file/store/selector/file.selector.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createSelector } from 'reselect'
+import type { DeepReadonly } from 'superTypes'
+import type { AppState, FileById } from '../appState'
+import type { MediaType } from 'domain/common/type'
+import type { FileEntity } from 'domain/file/entity/file'
+
+export type FileDescriptor = {
```
Comment (fredericvilcot) [inline src/domain/file/store/selector/file.selector.ts:7]: We also need the `id` property here that must be mandatory.

File: src/domain/file/store/selector/file.selector.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  readonly name: string
+  readonly size: number
+  readonly type: MediaType
+}
+
+export const getFiles: (state: DeepReadonly<AppState>) => FileDescriptor[] = createSelector(
+  (state: DeepReadonly<AppState>): FileById => state.file.byId,
+  (files: DeepReadonly<FileById>): FileDescriptor[] => [
+    ...files.map(({ name, size, type }: FileEntity) => ({ name, size, type })).toList()
```
Comment (fredericvilcot) [inline src/domain/file/store/selector/file.selector.ts:16]: I think here we can avoid the use of the `immutable` `map` function that returns a new `OrderedMap` which weighs down the function for nothing. 
IMHO, It's better to first convert the `OrderedMap` into an JS `Array` and then `map` on it.
```suggestion
    files
      .toIndexedSeq()
      .toArray()
      .map(({ id, name, size, type }: FileEntity) => ({ id, name, size, type }))
```

File: src/domain/file/store/selector/file.selector.spec.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  file: {
+    byId: OrderedMap<string, FileEntity>().set(file1.id, file1).set(file2.id, file2),
+    byType: OrderedMap<string, OrderedSet<string>>()
+      .set(file1.type, OrderedSet<string>([file1.id]))
+      .set(file2.type, OrderedSet<string>([file2.id]))
+  }
+}
+
+const expectedFiles = (): FileDescriptor[] =>
+  [file1, file2].map(({ name, size, type }) => ({ name, size, type }))
```
Comment (fredericvilcot) [inline src/domain/file/store/selector/file.selector.spec.ts:41]: We need the `id` property in the `expectedFiles`

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
Hey, your first ever PR on domain business, great! 
I wrote down some misses, let me know if you need more explanations!

**Comment by lolottetheclash** [comment]
> Hey, your first ever PR on domain business, great! I wrote down some misses, let me know if you need more explanations!

Thanks for all your relevant remarks 👌I'll fix it right away

**Review by ErikssonJoakim — APPROVED** [review]

**Review by fredericvilcot — APPROVED** [review]
Great!

**Review by amimart — APPROVED** [review]
Seems good!

**Review by JoeyDjr — APPROVED** [review]

**Pull request merged**